go.sum entries not updated correctly for dependencies referenced only in tests

### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Describe the bug

Sample repo: https://github.com/charleskorn/renovate-golang-test-deps-issue
Sample PR: https://github.com/charleskorn/renovate-golang-test-deps-issue/pull/2

It seems that Renovate is not updating `go.sum` entries for dependencies that are only referenced in test files.

Running the tests for a PR affected by this issue results in an error like the following:

```
# github.com/charleskorn/renovate-golang-test-deps-issue
app_test.go:6:2: missing go.sum entry for module providing package github.com/onsi/gomega (imported by github.com/charleskorn/renovate-golang-test-deps-issue); to add:
	go get -t github.com/charleskorn/renovate-golang-test-deps-issue
FAIL	github.com/charleskorn/renovate-golang-test-deps-issue [setup failed]
FAIL
```

Running `go get -t ...` as suggested results in another line being added to `go.sum`:

```diff
diff --git a/go.sum b/go.sum
index 3ca2d35..3e58c9c 100644
--- a/go.sum
+++ b/go.sum
@@ -32,6 +32,7 @@ github.com/onsi/gomega v1.7.1/go.mod h1:XdKZgCCFLUoM/7CFJVPcG8C1xQ1AJ0vpAezJrB7J
 github.com/onsi/gomega v1.10.1/go.mod h1:iN09h71vgCQne3DLsj+A5owkum+a2tYe+TOCB1ybHNo=
 github.com/onsi/gomega v1.17.0 h1:9Luw4uT5HTjHTN8+aNcSThgH1vdXnmdJ8xIfZ4wyTRE=
 github.com/onsi/gomega v1.17.0/go.mod h1:HnhC7FXeEQY45zxNK3PPoIUhzk/80Xly9PcubAlGdZY=
+github.com/onsi/gomega v1.18.0 h1:ngbYoRctxjl8SiF7XgP0NxBFbfHcg3wfHMMaFHWwMTM=
 github.com/onsi/gomega v1.18.0/go.mod h1:0q+aL8jAiMXy9hbwj2mr5GziHiwhAIQpFmmtT5hitRs=
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
```

These lines are added for dependencies referenced in non-test (ie. production) code.



### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Setting current branch to main(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: latest commit(branch="renovate/github.com-onsi-gomega-1.x")
{
  "branchName": "main",
  "latestCommitDate": "2022-01-26T16:23:39+11:00",
  "sha": "fa7f20c6df077149d16be310629dff7c33df75b1"
}
DEBUG: getBranchPr(renovate/github.com-onsi-gomega-1.x)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: findPr(renovate/github.com-onsi-gomega-1.x, undefined, open)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: findPr(renovate/github.com-onsi-gomega-1.x, undefined, closed)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: branchExists=false(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: dependencyDashboardCheck=undefined(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: recreateClosed is false(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: findPr(renovate/github.com-onsi-gomega-1.x, Update module github.com/onsi/gomega to v1.18.0, !open)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: prAlreadyExisted=false(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Checking schedule(at any time, null)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: No schedule defined(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Branch needs creating(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Using reuseExistingBranch: false(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: gomod.updateDependency: v1.18.0(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Updating github.com/onsi/gomega in go.mod(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: gomod.updateArtifacts(go.mod)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Adding Git authentication for Go Module retrieval for https://api.github.com/ using token auth.(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: go get command included(branch="renovate/github.com-onsi-gomega-1.x")
{
  "cmd": "go",
  "args": "get -d ./..."
}
DEBUG: Using docker to execute(branch="renovate/github.com-onsi-gomega-1.x")
{
  "image": "go"
}
DEBUG: Found version constraint - checking for a compatible image to use(branch="renovate/github.com-onsi-gomega-1.x")
{
  "depName": "docker.io/renovate/go",
  "scheme": "npm",
  "constraint": "^1.17"
}
DEBUG: getLabels(https://index.docker.io, renovate/go, latest)(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Found compatible image version(branch="renovate/github.com-onsi-gomega-1.x")
{
  "depName": "docker.io/renovate/go",
  "scheme": "npm",
  "constraint": "^1.17",
  "version": "1.17.6"
}
DEBUG: Resolved tag constraint(branch="renovate/github.com-onsi-gomega-1.x")
{
  "image": "docker.io/renovate/go",
  "tagConstraint": "^1.17",
  "tagVersioning": "npm",
  "tag": "1.17.6"
}
DEBUG: Fetching Docker image: docker.io/renovate/go:1.17.6(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Finished fetching Docker image(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Executing command(branch="renovate/github.com-onsi-gomega-1.x")
{
  "command": "docker run --rm --name=renovate_go --label=renovate_child -v \"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\":\"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e GOPATH -e GOPROXY -e GOFLAGS -e CGO_ENABLED -e GIT_CONFIG_KEY_0 -e GIT_CONFIG_VALUE_0 -e GIT_CONFIG_COUNT -e GIT_CONFIG_KEY_1 -e GIT_CONFIG_VALUE_1 -w \"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\" docker.io/renovate/go:1.17.6 bash -l -c \"go get -d ./...\""
}
DEBUG: exec completed(branch="renovate/github.com-onsi-gomega-1.x")
{
  "cmd": "docker run --rm --name=renovate_go --label=renovate_child -v \"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\":\"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e GOPATH -e GOPROXY -e GOFLAGS -e CGO_ENABLED -e GIT_CONFIG_KEY_0 -e GIT_CONFIG_VALUE_0 -e GIT_CONFIG_COUNT -e GIT_CONFIG_KEY_1 -e GIT_CONFIG_VALUE_1 -w \"/mnt/renovate/gh/charleskorn/renovate-golang-test-deps-issue\" docker.io/renovate/go:1.17.6 bash -l -c \"go get -d ./...\"",
  "durationMs": 3065,
  "stdout": "",
  "stderr": ""
}
DEBUG: Returning updated go.sum(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Updated 1 package files(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Updated 1 lock files(branch="renovate/github.com-onsi-gomega-1.x")
{
  "updatedArtifacts": [
    "go.sum"
  ]
}
DEBUG: 2 file(s) to commit(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Committing files to branch renovate/github.com-onsi-gomega-1.x(branch="renovate/github.com-onsi-gomega-1.x")
DEBUG: Setting git author name(branch="renovate/github.com-onsi-gomega-1.x")
{
  "gitAuthorName": "Renovate Bot"
}
DEBUG: Setting git author email(branch="renovate/github.com-onsi-gomega-1.x")
{
  "gitAuthorEmail": "bot@renovateapp.com"
}
DEBUG: git commit(branch="renovate/github.com-onsi-gomega-1.x")
{
  "deletedFiles": [],
  "ignoredFiles": [],
  "result": {
    "author": null,
    "branch": "renovate/github.com-onsi-gomega-1.x",
    "commit": "cafe11f",
    "root": false,
    "summary": {
      "changes": 2,
      "insertions": 10,
      "deletions": 1
    }
  }
}
```

</details>


### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

--- a/lib/manager/gomod/artifacts.ts
+++ b/lib/manager/gomod/artifacts.ts
@@ -184,7 +184,7 @@ export async function updateArtifacts({
 
     const execCommands = [];
 
-    let args = 'get -d ./...';
+    let args = 'get -d -t ./...';
     logger.debug({ cmd, args }, 'go get command included');
     execCommands.push(`${cmd} ${args}`);
 

